src/dynamic_boundary_conditions/tide/sea_level_rise_data.py

```diff
@@ -145,28 +92,64 @@ def read_slr_data_from_files(slr_data_dir: pathlib.Path) -> gpd.GeoDataFrame:
+
     # Create an empty list to store the sea level rise datasets
-    slr_nz_list = []
-    # Loop through each CSV file in the specified directory
-    for file_path in slr_data_dir.glob("*.csv"):
-        # Read the CSV file into a pandas DataFrame using pyarrow
-        slr_region = csv.read_csv(file_path).to_pandas()
-        # Extract the region name from the file name and add it as a new column in the DataFrame
-        file_name = file_path.stem
-        start_index = file_name.find('projections_') + len('projections_')
-        end_index = file_name.find('_region')
-        region_name = file_name[start_index:end_index]
-        slr_region['region'] = region_name
-        # Append the DataFrame to the list
-        slr_nz_list.append(slr_region)
-        # Log that the file has been successfully loaded
-        log.info(f"Successfully loaded the '{file_path.name}' data file.")
-    # Concatenate all the dataframes in the list and add geometry column
-    slr_nz = pd.concat(slr_nz_list, axis=0).reset_index(drop=True)
-    geometry = gpd.points_from_xy(slr_nz['lon'], slr_nz['lat'], crs=4326)
-    slr_nz_with_geom = gpd.GeoDataFrame(slr_nz, geometry=geometry)
-    # Convert all column names to lowercase
-    slr_nz_with_geom.columns = slr_nz_with_geom.columns.str.lower()
+    slr_nz_list = [
+        pd.read_csv(file_path) for file_path in slr_data_dir.glob("*.csv")
+    ]
+
+    # Merge Site Details dataframe and Sea level projections tables with VLM
+    slr_nz_merge_vlm = pd.merge(
+        slr_nz_list[0], slr_nz_list[1], left_on='Site ID', right_on='site',
+        how='left',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_vlm['add_vlm'] = True
+
+    # Merge Site Details dataframe and Sea level projections tables with VLM
+    slr_nz_merge_novlm = pd.merge(
+        slr_nz_list[0], slr_nz_list[2], left_on='Site ID', right_on='site',
+        how='left',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_novlm['add_vlm'] = False
+
+    # Concatenate all the dataframes in the list
+    slr_nz = pd.concat([slr_nz_merge_vlm, slr_nz_merge_novlm], axis=0).reset_index(drop=True)
+
+    # Remove unnamed columns
+    slr_nz_df = slr_nz.loc[:, ~slr_nz.columns.str.contains('^Unnamed')]
+    # Remove site column
+    slr_nz_df = slr_nz_df.drop(columns=['site'])
+    # Rename the columns
+    slr_nz_df = slr_nz_df.rename(columns={
+        'Site ID': 'siteid',
+        'Lon': 'lon',
+        'Lat': 'lat',
+        'Vertical Rate (mm/yr)': 'vertical_rate',
+        'Vertical Rate - BOP corrected (mm/yr)': 'vertical_rate_bop',
+        '1-sigma uncertainty (mm/yr)': 'sigma_uncertainty',
+        'Number of obs': 'number_of_obs',
+        'Quality Factor': 'quality_factor',
+        'Average distance between coastal point and observations': "average_distance",
+        'Confidence': 'confidence_level',
+        '0.17': 'p17',
+        '0.5': 'p50',
+        '0.83': 'p83',
+        'SSP': 'ssp'
+    })
```

Comment:
It looks like the column 'year' is missing.

Reply:
Ah the year column is in the dataset already, so I did not modify it here.